### PR TITLE
[FEATURE] Implement Dark Mode Toggle and Theme Support

### PR DESCRIPTION
**Pull request by ARRY7686**
### Description 
This feature introduces a dark mode toggle button, allowing users to switch between light and dark themes. The theme state is persisted across sessions, ensuring a consistent user experience. The implementation includes:

- A toggle button for switching between light and dark modes.
- Theme persistence using local storage (or other methods if applicable).
- Aesthetic adjustments for both light and dark themes to enhance readability and usability.

### Tasks:

- Ensure the toggle button functionality works as expected.
- Validate the correct application of styles for both themes.
- Confirm theme persistence across sessions.
- Test the feature on different devices and screen sizes.

### CODE REVIEW

**Comment by ARRY7686** [comment]
@manuelalferez Have successfully made the dark mode toggle and added it to the navbar thanks for trusting and assigning me with issue #62 had fun working on this project and looking forward to more contributions here.

Please add the hacktoberfest-accepted label and have a great day ahead

File: .env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+NEXT_PUBLIC_SUPABASE_URL="https://rmyvutrqvavmemptyxwb.supabase.co"
+NEXT_PUBLIC_SUPABASE_ANON_KEY=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InJteXZ1dHJxdmF2bWVtcHR5eHdiIiwicm9sZSI6ImFub24iLCJpYXQiOjE3Mjg0MzI4NTYsImV4cCI6MjA0NDAwODg1Nn0.gNMrbihw3QfPfQxgrziUE9Vx0LXmMUqqP3QgvbPc1Tw
```
Comment (manuelalferez) [inline .env:2]: Oops! You should rotate your `anon` key, as it's not safe to publish your environment variables. In Next.js, local environment variables are stored in the `.env.local` file.

Comment (manuelalferez) [reply .env:2]: `.env` file shouldn't be part of the commit

**Comment by manuelalferez** [comment]
In the landing page, this is not working well with dark mode: 

image

**Comment by ARRY7686** [comment]
Ahh I see I guess some styles are not applied I will look into that

**Comment by ARRY7686** [comment]
hey @manuelalferez fixed the issues please take a look

**Comment by jesi-rgb** [comment]
@ARRY7686 there are still some problems! i found out upon logging in

image

**Comment by manuelalferez** [comment]
I see the improvements!

Could you please arrange this ones: 
image

image

image

image

image

It's getting closer to merge it. Everyone will love this feature

**Comment by manuelalferez** [comment]
@ARRY7686 any update on this?

**Comment by ARRY7686** [comment]
@manuelalferez yeah I didn't get the time as I had my university exams I will update it and get back to you in this week as I can see there are some conflicts that have arised.

**Comment by ARRY7686** [comment]
Thanks a lot

**Pull request closed without merging**